### PR TITLE
Calling .leave request when unsubscribing

### PR DESCRIPTION
**Pull request by jguz-pubnub**
fix(presence): calling .leave request in missing cases

### CODE REVIEW

File: Sources/PubNub/EventEngine/Presence/PresenceTransition.swift [hunk 1]
```diff
@@ -26,15 +26,15 @@ class PresenceTransition: TransitionProtocol {
...
+      return true
     case .heartbeatSuccess:
       return state is Presence.Heartbeating
     case .heartbeatFailed:
       return state is Presence.Heartbeating
     case .timesUp:
       return state is Presence.HeartbeatCooldown
     case .leftAll:
-      return !(state is Presence.HeartbeatInactive)
+      return true
```
Comment (parfeon) [inline Sources/PubNub/EventEngine/Presence/PresenceTransition.swift:37]: So, we want to transit even if heartbeat loop is not active?

**Review by parfeon — APPROVED** [review]
LGTM!

File: Sources/PubNub/EventEngine/Presence/PresenceTransition.swift (hunk 1 above)
Comment (jguz-pubnub) [reply Sources/PubNub/EventEngine/Presence/PresenceTransition.swift:37]: Yes. You're right, the heartbeat loop is inactive if the user subscribes to some channel without Presence. However, we still have to trigger `.leave` invocation as a part of the unsubscribe cleanup

**Comment by jguz-pubnub** [comment]
@pubnub-release-bot release as 7.3.1

**Pull request merged**